fastar/rules.py

```diff
@@ -12,6 +12,22 @@
 map = safe_map
 zip = safe_zip
 
+@curry
+def default_update_rule(backward_rule, fun, cache, outbox, *args, **kwargs):
+  out_start, out_shape = outbox
+  instarts, incounts = backward_rule(outbox, *args, **kwargs)
+  inslices = [None if instart is None else
+              lax.dynamic_slice(arg, instart, incount.shape)
+              for arg, instart, incount in zip(args, instarts, incounts)]
+  out_slice = fun(*inslices, **kwargs)
+  return lax.dynamic_update_slice(cache, out_slice, out_start)
+
+def def_standard_op(op, backward_rule, slice_fun=None):
```

Comment:
For now can we simplify this by removing the slice_fun kwarg and only applying this rule when the slice fun is op.bind?

Although this will lead to longer rules for ops like concatenate, I think it makes it slightly easier for a new reader to understand the code.

Reply:
Ok, done!